examples/lines_bars_and_markers/timeline.py

```diff
@@ -15,57 +15,98 @@
+except:
+    # In case the above fails, e.g. because of missing internet connection
+    # use the following lists as fallback.
+    names = ['v2.2.4', 'v3.0.3', 'v3.0.2', 'v3.0.1', 'v3.0.0', 'v2.2.3',
+             'v2.2.2', 'v2.2.1', 'v2.2.0', 'v2.1.2', 'v2.1.1', 'v2.1.0',
+             'v2.0.2', 'v2.0.1', 'v2.0.0', 'v1.5.3', 'v1.5.2', 'v1.5.1',
+             'v1.5.0', 'v1.4.3', 'v1.4.2', 'v1.4.1', 'v1.4.0']
+
+    dates = ['2019-02-26', '2019-02-26', '2018-11-10', '2018-11-10',
+             '2018-09-18', '2018-08-10', '2018-03-17', '2018-03-16',
+             '2018-03-06', '2018-01-18', '2017-12-10', '2017-10-07',
+             '2017-05-10', '2017-05-02', '2017-01-17', '2016-09-09',
+             '2016-07-03', '2016-01-10', '2015-10-29', '2015-02-16',
+             '2014-10-26', '2014-10-18', '2014-08-26']
+
+# Convert date strings (e.g. 2014-10-18) to datetime
+dates = [datetime.strptime(d, "%Y-%m-%d") for d in dates]
 
 ##############################################################################
-# Next, we'll iterate through each date and plot it on a horizontal line.
-# We'll add some styling to the text so that overlaps aren't as strong.
+# Next, we'll create a `~.Axes.stem` plot with some variation in levels as to
+# distinguish even close-by events.
+# For each event, we add a text label via `~.Axes.annotate`, which is offset
+# in units of points from the tip of the event line.
 #
 # Note that Matplotlib will automatically plot datetime inputs.
 
-levels = np.array([-5, 5, -3, 3, -1, 1])
-fig, ax = plt.subplots(figsize=(8, 5))
-
-# Create the base line
-start = min(dates)
-stop = max(dates)
-ax.plot((start, stop), (0, 0), 'k', alpha=.5)
-
-# Iterate through releases annotating each one
-for ii, (iname, idate) in enumerate(zip(names, dates)):
-    level = levels[ii % 6]
-    vert = 'top' if level < 0 else 'bottom'
-
-    ax.scatter(idate, 0, s=100, facecolor='w', edgecolor='k', zorder=9999)
-    # Plot a line up to the text
-    ax.plot((idate, idate), (0, level), c='r', alpha=.7)
-    # Give the text a faint background and align it properly
-    ax.text(idate, level, iname,
-            horizontalalignment='right', verticalalignment=vert, fontsize=14,
-            backgroundcolor=(1., 1., 1., .3))
+
+# Choose some nice levels
+levels = np.tile([-5, 5, -3, 3, -1, 1],
+                 int(np.ceil(len(dates)/6)))[:len(dates)]
+
+# Create figure and plot a stem plot with the date
+fig, ax = plt.subplots(figsize=(8.8, 4), constrained_layout=True)
 ax.set(title="Matplotlib release dates")
-# Set the xticks formatting
-# format xaxis with 3 month intervals
-ax.get_xaxis().set_major_locator(mdates.MonthLocator(interval=3))
+
+markerline, stemline, baseline = ax.stem(dates, levels,
+                                         linefmt="C3-", basefmt="k-",
+                                         use_line_collection=True)
+
+plt.setp(markerline, mec="k", mfc="w", zorder=3)
+markerline.set_ydata(np.zeros(len(dates)))
```

Comment:
This is a hack on the original purpose of `stem`. Not sure I like it, but OTOH, the loop before wasn't good either. At least, this needs explanation.